### PR TITLE
196 refacto create entities functions

### PR DESCRIPTION
**Pull request by osvegn**
## Update features
- RTypeLogicCreateEntitiesFunctions library is now a static library. It isn't possible to do a shared library because of a cycle dependence with other library.
- Create entities functions have been moved in .cpp file
- Remove link with RTypeLogicCreateEntitiesFunctions in others library if it isn't useful

### CODE REVIEW

File: libs/R-TypeLogic/EntityManipulation/CreateEntitiesFunctions/CreateAlliedProjectile.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        Position pos = ally.getComponent<Position>();
+        Damage damage = ally.getComponent<Damage>();
+        Velocity velocity = ally.getComponent<Velocity>();
+
+        Entity &entity = world.addEntity()
+                             .addComponent<Position>(pos.x + 102, pos.y + 45)
+                             .addComponent<Velocity>(300, 0)
+                             .addComponent<Weight>(1)
+                             .addComponent<Size>(40, 40)
+                             .addComponent<LifeTime>(100)
```
Comment (31Nathan) [inline libs/R-TypeLogic/EntityManipulation/CreateEntitiesFunctions/CreateAlliedProjectile.cpp:24]: LIfetime is now at 3, not important bc waiting for the lifetime bug fix

File: libs/R-TypeLogic/EntityManipulation/CreateEntitiesFunctions/CreateEnemyProjectile.cpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    {
+        Position pos = enemy.get()->getComponent<Position>();
+        Damage damage = enemy.get()->getComponent<Damage>();
+
+        Entity &entity = world.addEntity()
+                             .addComponent<Position>(pos.x, pos.y + 20)
+                             .addComponent<Velocity>(-400, 0)
+                             .addComponent<Weight>(1)
+                             .addComponent<Size>(40, 40)
+                             .addComponent<LifeTime>(100)
```
Comment (31Nathan) [inline libs/R-TypeLogic/EntityManipulation/CreateEntitiesFunctions/CreateEnemyProjectile.cpp:23]: Same

File: libs/R-TypeLogic/EntityManipulation/CreateEntitiesFunctions/CreateObstacle.cpp [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+{
+
+    std::size_t createNewObstacle(World &world, const int posX, const int posY, const unsigned short damage,
+        const std::string uuid, unsigned short networkId)
+    {
+        Entity &entity = world.addEntity()
+                             .addComponent<Position>(posX, posY)
+                             .addComponent<Weight>(1)
+                             .addComponent<Size>(1, 1)
+                             .addComponent<LifeTime>()
```
Comment (31Nathan) [inline libs/R-TypeLogic/EntityManipulation/CreateEntitiesFunctions/CreateObstacle.cpp:20]: Remove lifetime from Obstacle

**Review by 31Nathan — CHANGES_REQUESTED** [review]

File: libs/R-TypeLogic/EntityManipulation/CreateEntitiesFunctions/CreateAlliedProjectile.cpp (hunk 1 above)
Comment (osvegn) [reply libs/R-TypeLogic/EntityManipulation/CreateEntitiesFunctions/CreateAlliedProjectile.cpp:24]: Done

File: libs/R-TypeLogic/EntityManipulation/CreateEntitiesFunctions/CreateEnemyProjectile.cpp (hunk 2 above)
Comment (osvegn) [reply libs/R-TypeLogic/EntityManipulation/CreateEntitiesFunctions/CreateEnemyProjectile.cpp:23]: Done

File: libs/R-TypeLogic/EntityManipulation/CreateEntitiesFunctions/CreateObstacle.cpp (hunk 3 above)
Comment (osvegn) [reply libs/R-TypeLogic/EntityManipulation/CreateEntitiesFunctions/CreateObstacle.cpp:20]: Done

**Review by LucasTesnier — APPROVED** [review]

**Pull request merged**